plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/server/ServiceViewModel.java

```diff
@@ -0,0 +1,198 @@
+/*******************************************************************************
+ * Copyright (c) 2015 Red Hat, Inc.
+ * Distributed under license by Red Hat, Inc. All rights reserved.
+ * This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution,
+ * and is available at http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     Red Hat, Inc. - initial API and implementation
+ ******************************************************************************/
+package org.jboss.tools.openshift.internal.ui.server;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import org.jboss.tools.common.databinding.ObservablePojo;
+import org.jboss.tools.openshift.common.core.connection.ConnectionsRegistrySingleton;
+import org.jboss.tools.openshift.core.connection.Connection;
+import org.jboss.tools.openshift.internal.ui.treeitem.IModelFactory;
+import org.jboss.tools.openshift.internal.ui.treeitem.ObservableTreeItem;
+
+import com.openshift.restclient.ResourceKind;
+import com.openshift.restclient.model.IProject;
+import com.openshift.restclient.model.IService;
+
+/**
+ * @author Andre Dietisheim
+ */
+public class ServiceViewModel extends ObservablePojo {
+
+	public static final String PROPERTY_CONNECTION = "connection";
+	public static final String PROPERTY_CONNECTIONS = "connections";
+	public static final String PROPERTY_SERVICE = "service";
+	public static final String PROPERTY_SERVICE_ITEMS = "serviceItems";
+	
+	private Connection connection;
+	private List<Connection> connections = new ArrayList<>();
+	private List<ObservableTreeItem> serviceItems = new ArrayList<>();
+	private IService service;
+
+	public ServiceViewModel(Connection connection) {
+		this(null, connection);
+	}
+
+	public ServiceViewModel(IService service, Connection connection) {
+		this.connection = connection;
+		this.service = service;
+	}
+
+	protected void update(Connection connection, List<Connection> connections, IService service, List<ObservableTreeItem> serviceItems) {
+		updateConnections(connections);
+		updateConnection(connection);
+		updateServiceItems(serviceItems);
+		updateService(service, serviceItems);
+	}
+
+	protected void updateConnection(Connection connection) {
+		firePropertyChange(PROPERTY_CONNECTION, this.connection, this.connection = connection);
+	}
+
+	private void updateConnections(List<Connection> connections) {
+		List<Connection> oldItems = new ArrayList<>(this.connections);
+		// ensure we're not operating on the same list
+		List<Connection> newItems = new ArrayList<>();
+		if (connections != null) {
+			newItems.addAll(connections);
+		}
+		this.connections.clear();
+		this.connections.addAll(newItems);
+		firePropertyChange(PROPERTY_CONNECTIONS, oldItems, this.connections);
```

Comment:
If concern is that 'this.connections' and 'connections' may be the same object, code still may be more concise:

List<Connection> oldItems = new ArrayList<>(this.connections);
if(connections != this.connections) {
    this.connections.clear();
    if(connections !=null) {
        this.connections.addAll(connections);
    }
}

Besides, if 'connections' and 'this.connections' is the same object then there is no need to fire so that check if(connections != this.connections) can wrap the entire method content, not just copying list.